src/test/java/reactor/pool/CommonPoolTest.java

```diff
@@ -2167,4 +2163,134 @@ void poolCloseRacingWithRelease(Function<PoolBuilder<AtomicInteger, ?>, Abstract
 		}
 	}
 
+	@ParameterizedTest
+	@MethodSource("allPools")
+	void raceShutdownAndAcquireInvalidate(Function<PoolBuilder<AtomicInteger, ?>, AbstractPool<AtomicInteger>> configAdjuster) {
+		AtomicInteger ai = new AtomicInteger();
+		PoolBuilder<AtomicInteger, PoolConfig<AtomicInteger>> configBuilder = PoolBuilder
+				.from(Mono.fromSupplier(() -> {
+					ai.incrementAndGet();
+					return ai;
+				}))
+				.evictionIdle(Duration.ZERO)
+				.destroyHandler(resource -> Mono.fromRunnable(resource::decrementAndGet))
+				.sizeBetween(0, 1);
+
+		AtomicReference<Throwable> errorRef = new AtomicReference<>();
+		for (int i = 0; i < 100_000; i++) {
```

Comment:
Just an idea: looks like a good use case for JCStress :)

Reply:
yeah I know, I'll need to convert these to JCStress at one point